### PR TITLE
A new era

### PR DESCRIPTION
**Pull request by AffectedArc07**
# Intro

TLDR - The development structure is getting a massive shakeup 

## Main Post

The PR approval process has had a massive shift, with heads + ~~maints~~ headcoders no longer being the only say on decisions.

The development team now has 2 specific groups for votes, the `Balance Team` and the `Design Team`. PR votes will now go as follows.

---

#### i - ALL PRs

- All PRs are subject to a 30-day timeout, unless:
  - It is pending testmerge.
  - It is waiting for the PR author to make requested changes.
    - If the PR author has not responded to the request for changes within `7` days, the PR is to be marked stale and closed.
- If a PR has passed design approval and is waiting on code review, the reviewers have 14 days to look at it before it is handed to a `Headcoder`.
  - This means `14` days for someone to even mention they are reviewing it. It does not mean that you must complete the PR review within 14 days.
- If a vote is sat at a 50/50 deadlock for whatever reason, a `Headcoder` must vote to override the deadlock.
  - The `Headcoder` may **not** vote twice to solve a deadlock if they have already voted in this matter.

#### ii - Pure Fix/Refactor PR with no design implications

- PR is to be reviewed by the code review them then merged by commit access. No waiting period is necessary.

#### iii - Content expansion (Major feature PR)

- Design is first looked over by the `Design Team`, ensuring it reaches majority vote.
  - This preferably happens before the contributor puts hours of work in.
  - Because of this vote, the PR must stay open for at least 24 hours
- The `Balance Team` gives it a skim over for glaring balance issues, but they **cannot** throw a merge-blocking objection on it.
  - They can however, speak to the `Design Team` to try and make their point reasonably.
- Voting then occurs again after any design/balance changes have been requested to make sure its still agreed upon.
- Once the PR has passed the design vote, it is then handed to the `Review Team` to ensure the code is of sufficient quality.
- Once the review team has ensured quality, a member of `Commit Access` can merge the PR.

#### iv - Minor balance change (Value tweaking)

- The PR is looked over by the `Balance Team` only, and voting occurs.
  - Because of this vote, the PR must stay open for at least 24 hours.
- Once the PR has passed the balance vote, it is then handed to the `Review Team` to ensure the code is of sufficient quality.
- Once the review team has ensured quality, a member of `Commit Access` can merge the PR.

#### v - Major balance change (New weapon or ERT class or similar)

- The PR is looked over by the `Balance Team` first to refine the values, and voting occurs.
  - Because of this vote, the PR must stay open for at least 24 hours.
- The `Design Team` then looks at it to ensure it fits in with the game design.
  - In a similar fashion to before, the `Design Team` themselves **cannot** forcefully merge-block the PR, but they can speak to the `Balance Team` reasonably.
- Once the PR has passed the balance vote, it is then handed to the `Review Team` to ensure the code is of sufficient quality.
- Once the review team has ensured quality, a member of `Commit Access` can merge the PR.

#### vi - Non-balance altering tweak

- The PR is looked over by the `Design Team` only, and voting occurs.
  - Because of this vote, the PR must stay open for at least 24 hours.
- Once the PR has passed the balance vote, it is then handed to the `Review Team` to ensure the code is of sufficient quality.
- Once the review team has ensured quality, a member of `Commit Access` can merge the PR.

#### vii - Any PR that doesn't fit above criteria

- At the discretion of `Headcoders`.

---

The `Design Team` also has the responsibility of managing projects and a general roadmap for the server. This includes the TGUI conversions, sprite reworks, ways to spice up departments, general direction of the game, and anything else they think of.

---

PR status will now also be communicated with new labels. These will be

- `Status: Awaiting type assignment` - This will be displayed when your PR is awaiting an internal type assignment (for Fix, Balance, Tweak, etc)
- `Status: Awaiting approval` - This will be displayed if your PR is waiting for approval from the specific party, be it Balance or Design. Fixes & Refactors should never have this label
- `Status: Awaiting review` - This will be displayed when your PR has passed the design vote and is now waiting for someone in the review team to approve it
- `Status: Awaiting merge` - Your PR is done and is waiting for someone with commit access to merge it. **Note: Your PR may be delayed if it is pending testmerge or in the mapping queue** 

---

If you have any questions about this reform, do not be afraid to ask me.

### CODE REVIEW

**Comment by KittyNoodle** [comment]
video games

**Review by S34NW — APPROVED** [review]
Grammar is robust

**Review by SteelSlayer — APPROVED** [review]

**Review by hal9000PR — APPROVED** [review]
